nomos-cli/src/da/disseminate.rs

```diff
@@ -1,76 +1,79 @@
-use crate::api::mempool::send_certificate;
-use clap::{Args, ValueEnum};
-use full_replication::{AbsoluteNumber, Attestation, Certificate, FullReplication, Voter};
-use futures::StreamExt;
-use hex::FromHex;
-use nomos_core::wire;
-use nomos_da::network::{adapters::libp2p::Libp2pAdapter as DaNetworkAdapter, NetworkAdapter};
+use clap::Args;
+use kzgrs_backend::{
+    common::build_attestation_message,
+    dispersal::{BlobInfo, Metadata},
+    encoder::EncodedData as KzgEncodedData,
+};
+use nomos_core::{
+    da::{DaDispersal, DaEncoder},
+    wire,
+};
+use nomos_da_network_service::{
+    backends::mock::executor::MockExecutorBackend as NetworkBackend, NetworkService,
+};
 use nomos_log::Logger;
-use nomos_network::backends::libp2p::Libp2p as NetworkBackend;
-use nomos_network::NetworkService;
 use overwatch_derive::*;
 use overwatch_rs::{
     services::{
         handle::{ServiceHandle, ServiceStateHandle},
-        relay::NoMessage,
+        relay::{NoMessage, OutboundRelay, Relay},
         state::*,
         ServiceCore, ServiceData, ServiceId,
     },
     DynError,
 };
 use reqwest::Url;
-use serde::Serialize;
 use std::{
-    error::Error,
     path::PathBuf,
     sync::{mpsc::Sender, Arc},
     time::Duration,
 };
 use tokio::sync::{mpsc::UnboundedReceiver, Mutex};
 
-pub async fn disseminate_and_wait<D, B, N, A, C>(
-    mut da: D,
+use super::network::adapters::mock::MockExecutorDispersalAdapter;
+use crate::api::mempool::send_blob_info;
+
+const APP_ID: &[u8; 32] = b"1234567890abcdef1234567890abcdef";
```

Comment:
lol
We could add something meaningful though, like: `32bytes_hash("nomos-cli")`.

Reply:
Moving this to app settings, will be passed via vars.